app/src/main/java/pm/gnosis/heimdall/ui/onboarding/RestoreAccountViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package pm.gnosis.heimdall.ui.onboarding
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import pm.gnosis.heimdall.accounts.base.repositories.AccountsRepository
+import javax.inject.Inject
+
+class RestoreAccountViewModel @Inject constructor(private val accountsRepository: AccountsRepository) : RestoreAccountContract() {
+    override fun isValidMnemonic(mnemonic: String): Single<Boolean> {
+        val words = mnemonic.split(Regex("\\s+"))
+        if (words.size < 12) return Single.just(false)
```

Comment:
Also we could return throw an exception if invalid to pass along some information why it is invalid

Reply:
Since a validation function would return true if it was valid and an exception if it was not valid (instead of false) I decided to go with a `sealed class` implementation.